lisa/sut_orchestrator/azure/common.py

```diff
@@ -132,10 +148,84 @@ def marketplace(self, value: Optional[AzureVmMarketplaceSchema]) -> None:
         else:
             self.marketplace_raw = value.to_dict()  # type: ignore
 
+    @property
+    def shared_gallery(self) -> Optional[SharedImageGallerySchema]:
+        # this is a safe guard and prevent mypy error on typing
+        if not hasattr(self, "_shared_gallery"):
+            self._shared_gallery: Optional[SharedImageGallerySchema] = None
+        shared_gallery: Optional[SharedImageGallerySchema] = self._shared_gallery
+        if not shared_gallery:
```

Comment:
If it's there, return. Why do people love needless indentation so much? :( This must be almost hitting https://www.flake8rules.com/rules/C901.html. BTW, time to clean that one from our code base, IMO--I also hit that once on my (internal) code and I loved it (very handy advice that you're really screwing up in legibility).

Reply:
Nice suggestion! Currently complexity is set to 15 but maybe we can bring it down to 10. These are the error I see if I reduce it to 10 : 
```
.\lisa\runner.py:235:5: C901 'RootRunner._start_loop' is too complex (15)
.\lisa\search_space.py:97:5: C901 'IntRange.check' is too complex (12)
.\lisa\search_space.py:299:1: C901 'check_countspace' is too complex (13)
.\lisa\runners\legacy_runner.py:629:1: C901 '_track_progress' is too complex (13)
.\lisa\sut_orchestrator\azure\platform_.py:601:5: C901 'AzurePlatform._get_environment_information' is too complex (15)
.\lisa\sut_orchestrator\azure\platform_.py:728:5: C901 'AzurePlatform._get_location_info' is too complex (11)
.\lisa\sut_orchestrator\azure\platform_.py:809:5: C901 'AzurePlatform._create_deployment_parameters' is too complex (15)
.\lisa\sut_orchestrator\azure\platform_.py:1183:5: C901 'AzurePlatform._resource_sku_to_capability' is too complex (11)
.\lisa\tools\reboot.py:50:5: C901 'Reboot.reboot' is too complex (12)
.\lisa\util\process.py:63:5: C901 'Process.start' is too complex (12)
.\lisa\util\process.py:144:5: C901 'Process.wait_result' is too complex (13)
```

The codebase is increasing and we can dedicate time to code quality improvements. @squirrelsc @glima Thoughts?

Reply:
Where is this set? I thought 10 was already the default? Yes, I vote for 10. Code quality should not be a pause/pending thing--ever, IMO. You do it as you go.

Reply:
It's set in .flake8 file under `max-complexity`

Reply:
We can set it to 10, but leave exception for current violations.

Reply:
I see. Please don't, @somil55. Either we fix that or leave for the next good willing one.

Reply:
15 is not too bad, I mean. We should fix that offender, though :/

Reply:
I wasn't planning on changing this as part of this PR. But I will add a issue to so that we take it up in future.

Reply:
Added issue https://github.com/microsoft/lisa/issues/1423 to track this